### DISCUSSION TITLE
GPU not working for Detectron2 - Needs changing 'cpu' to 'cuda' inside config.json

### DISCUSSION THREAD

**Discussion by AliButtarPG** [obss/sahi#642] [Q&A] [answered]
GPU wasn't working for predictions made using SAHI even though I was using `device` as `cuda`. I thought this setting would override the setting inside the configuration file regardless of whatever it was but once I changed the device to `cuda` inside the config.json then it worked like a charm. 

Something you could fix in the next iteration maybe. Thanks for the amazing tool.

**Comment by fcakyon** [comment]
Hello @AliButtarPG, thanks a lot for the bug report!

Can you be more specific on the issue?

Did I understand it correctly: 
1. you did not set `config_path` param
2. you set `device` param as cuda
3. then SAHI performs Detectron2 prediction on CPU?

**Comment by AliButtarPG** [comment]
Hi @fcakyon,

Sorry for not providing details, wasn't sure what more to write.

1. I did set `config_path` param and just gave path to the config of my custom model
2. Yes I set the `device` param as cuda like this: 
```
detection_model = Detectron2DetectionModel(
    model_path=model_path,
    config_path=config_path,
    confidence_threshold=0.1,
    device="cuda",  # or 'cuda:0'
)
```
3. Yes, until I changed the `device` param to cuda directly inside the `config.json`

**Comment by fcakyon** [comment]
@AliButtarPG we implemented SAHI Detectron2 support as: params in config file overwrites the params provided in Detectron2DetectionModel. This may be a bad practice, thanks for pointing that out! It would be better if the params in Detectron2DetectionModel would have higher priority 👍

**Comment by fcakyon** [comment, marked as answer]
This PR should fix this issue: https://github.com/obss/sahi/issues/478
